src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11118,8 +11118,7 @@ private LambdaExpressionSyntax ParseLambdaExpression(SyntaxToken asyncToken)
                     arrow = AddError(arrow, ErrorCode.ERR_NeedSpaceBetweenExclamationAndEquals);
                     var gt = this.EatToken(SyntaxKind.GreaterThanToken);
                     arrow = AddTrailingSkippedSyntax(arrow, gt);
-                    AddLeadingSkippedSyntax(arrow, notEq.GetLeadingTrivia());
-                    exclamation = AddLeadingSkippedSyntax(arrow, SyntaxFactory.MissingToken(SyntaxKind.ExclamationToken));
+                    exclamation = SyntaxFactory.MissingToken(SyntaxKind.ExclamationToken);
```

Comment:
👍   Hopefully, all the construction/shuffling of the raw information is clear here for you :)   I try to keep a mental model of all the nodes and what's going on with them, which helps make certain things stand out as looking "wrong".  i.e. we had the 'arrow' nodes appearing in two places.  on the 'arrow' local itself, but also as leading skipped syntax for the exclamation.  This violated the strict invariants of the tree.

Thanks!